macros/generic_tests/test_exists.sql

```diff
@@ -1,22 +1,34 @@
 {% macro test_exists(model, column_name) %}
 
+{% set table_exists = load_relation(model) is not none %}
+
 {# Test fails if 0 records are returned. #}
 {{ config(fail_calc = 'case when count(*) = 0 then 1 else 0 end') }}
 
-{# Query the column in INFORMATION_SCHEMA to check if it exists. #}
-select *
-from "INFORMATION_SCHEMA"."COLUMNS"
-where "INFORMATION_SCHEMA"."COLUMNS"."TABLE_SCHEMA" = '{{ model.schema }}'
-    and "INFORMATION_SCHEMA"."COLUMNS"."TABLE_NAME" = '{{ model.name }}'
-    and "INFORMATION_SCHEMA"."COLUMNS"."COLUMN_NAME" = replace('{{ column_name }}', '"', '')
+{# Only if the table exists or if we test on model level, we perform the real test. Otherwise a dummy is used #}
+{% if table_exists or column_name == null %}
+    {# Query the table info in INFORMATION_SCHEMA to check if it exists. If the test is defined on column level, also check if column exists #}
+    select *
+    from "INFORMATION_SCHEMA"."COLUMNS"
+    where "INFORMATION_SCHEMA"."COLUMNS"."TABLE_SCHEMA" = '{{ model.schema }}'
+        and "INFORMATION_SCHEMA"."COLUMNS"."TABLE_NAME" = '{{ model.name }}'
+        {%- if column_name != null -%}
+        and "INFORMATION_SCHEMA"."COLUMNS"."COLUMN_NAME" = replace('{{ column_name }}', '"', '')
+        {%- endif -%}
```

Comment:
It works. However, readability and understanding why it works is difficult.
What do you think of a structure as:

```
if (column name is not defined
 table check query
 set query
else if (column name is defined and Table exists
 column check query
 set query
```
The test becomes a bit longer, however expresses more direct what the use cases are.